src/assertions.cpp

```diff
@@ -218,6 +218,9 @@ class AssertExtractor {
             return { Assert{TypeConstraint{ins.dst, TypeGroup::number}} };
         }
     }
+    vector<Assert> operator()(IncrementLoopCounter ins) const {
+        assert(false);
+    }
```

Comment:
The windows build has an error saying something has to be returned, so maybe something like:
```suggestion
        return {};
    }
```